code/modules/reagents/chemistry/reagents/impure_reagents/impure_medicine_reagents.dm

```diff
@@ -67,12 +69,21 @@
 //Sends hands after you for your hubris
 /datum/reagent/inverse/helgrasp/on_mob_life(mob/living/carbon/owner, delta_time, times_fired)
 	spawn_hands(owner)
-	lag_compensate += 1 - max(delta_time, 1)
-	. = ..()
+	lag_remainder += delta_time - FLOOR(delta_time, 1)
+	delta_time = round(delta_time, 1)
+	if(lag_remainder >= 1)
+		delta_time += 1
+		lag_remainder -= 1
+	var/hands = 1
+	var/time = 2 / delta_time
+	while(hands < delta_time) //we already made a hand now so start from 1
+		LAZYADD(timer_id, addtimer(CALLBACK(src, .proc/spawn_hands, owner), (time*hands) SECONDS, TIMER_STOPPABLE)) //keep track of all the timers we set up
+		hands += time
```

Comment:
Could you explain what's going on here?

Reply:
Standard delta_time for a reagent is 2s - and volume consumption is equal to the volume * delta_time. In this chem, I want to consume 0.5u for 1 hand created (since 1*REM is 0.5) so on a single tick I create a hand and set up a callback for another one in 1s from now. But since delta time can vary, I want to be able to create more hands for when the delay is longer. 

Initally I round delta_time to the nearest whole number, and take the part that I am rounding down from (i.e. the decimal numbers) and keep track of them. If the decimilised numbers go over 1, then the number is reduced down by one and an extra hand is created that tick. (this _is_ how FLOOR works in byond, right? I'm pretty sure from testing.)

Then I attempt to calculate the how many hands to created based off the current delta_time, since I can't know the delay to the next one it assumes the next will be in 2s. I take the 2s interval period and divide it by the number of hands I want to make (i.e. the current delta_time) and I keep track of how many hands I'm creating (since I always create one on a tick, then I start at 1 hand). For each hand I then use this time value multiplied by the number of hands. Since we're spawning one now, and it checks to see if hands is less than, but not less than or equal to, delta_time, no hands will be created on the next expected tick. Basically, we fill the time between now and 2s from now with hands based off the current lag.

At the end, we clear up any loose hanging timers just in case and spawn any remaining lag_remaining hands all at once.